osf/models/preprint.py

```diff
@@ -951,6 +952,13 @@ def create_waterbutler_log(self, auth, action, payload):
             params=params
         )
 
```

Comment:
I am not sure which line you are referring to. Could you clarify?

Reply:
I couldn't exactly reference the line where `Preprint.add_contributor` isn't tagging the contributor's source... because there isn't one.

After reading over all this and the spec, I suspect the answer to my question might be either "Preprints don't [have all the weird legacy-type source tags] <OR> [try to pass `not user.is_registered` users to `add_contributor`], so the only time we would want to tag added contributors is in `add_unregistered_contributor`."

Reply:
Oh I think I know what you are referring to. in `Node.add_contributor()`, we only add system tags to users when the user added is not registered (unreg contrib). We do this because the legacy front-end uses the `add_contributor` method to add unreg contribs. For preprints, everything is in ember and uses V2. Therefore, adding a unreg contrib for preprint uses the `add_unregsitered_contributor` method. So we don't need to catch the case of adding unreg contrib in `Preprint.add_contributor()`.